src/segment-loader.js

```diff
@@ -975,6 +976,28 @@ export default class SegmentLoader extends videojs.EventTarget {
     // equal to the last appended mediaIndex
     if (this.mediaIndex !== null) {
       this.mediaIndex -= mediaSequenceDiff;
+
+      // this can happen if we are going to load the first segment, but get a playlist
+      // update during that. mediaIndex would go from 0 to -1 if mediaSequence in the
+      // new playlist was incremented by 1.
+      if (this.mediaIndex < 0) {
```

Comment:
I'm wondering if this will get us into trouble for short live playlists. Since we should start at the beginning of the playlist, the case may often happen that we end up with a mediaIndex of -1 while still processing the segment. In that case, the next segment we should load is segment 0 from the new playlist, which, if we do mediaIndex++, we'll be OK, unless we are at -2 or below. That may be a case for us to do a resync or seek. 

Do other areas of the code have trouble with a mediaIndex of -1?

Reply:
Don't we start at the end for live playlists?

Reply:
The reason that I added this code was: 
1. I have often seen issues with background tabs and live playlist refreshes happing way too slowly, so the segment we are on falls off of the playlist. 
2. having a negative media index doesn't really make sense, as we are dealing with zero based arrays. If the segment we are on falls off of the current playlist then the current media index would no longer be a valid media index for referring to the segment that we have appended.

I think ultimately the difference will be that we will never have a negative mediaIndex. So if we were two segments behind mediaIndex would not become -2 and leave us in a bad state. It would become negative and we would get the currentMediaIndex for time. 

It's possible that when mediaIndex and pendingSegment are both null, that we should probably try to jump closer to the live point in the stream. Right now we will just keep trying to request a segment at mediaIndex -1

Reply:
Looking again, I think we may be OK. My concern was mainly for the case of 3 segment live playlists. We start at the beginning of the first segment in the playlist, and it will probably be pretty often that we'll update our playlist during a segment request. If that happens, we clear out mediaIndex here (since the mediaIndex was 0 and we'd go to -1), so when we get to checkBuffer for the next segment,  we'd skip past mediaIndex++, but it looks like we'd still fetch at the end of buffer, so the guess should be OK: https://github.com/videojs/http-streaming/blob/82d83a70ea668a833ad1affef69a996d62383823/src/segment-loader.js#L1368-L1378